internal/backends/postgresql/database.go

```diff
@@ -50,7 +52,35 @@ func (db *database) ListCollections(ctx context.Context, params *backends.ListCo
 		return nil, lazyerrors.Error(err)
 	}
 
-	res := make([]backends.CollectionInfo, len(list))
+	var res []backends.CollectionInfo
+
+	if params != nil && len(params.Name) > 0 {
+
+		nameList := make([]string, len(list))
+		for i, c := range list {
+			nameList[i] = c.Name
+		}
+
+		res = make([]backends.CollectionInfo, 0, 1)
+
+		i, found := slices.BinarySearchFunc(nameList, params.Name, func(collectionName, t string) int {
+			return cmp.Compare(collectionName, t)
+		})
+
+		if found {
+			res = append(res, backends.CollectionInfo{
+				Name:            list[i].Name,
+				UUID:            list[i].UUID,
+				CappedSize:      list[i].CappedSize,
+				CappedDocuments: list[i].CappedDocuments,
+			})
+		}
+		return &backends.ListCollectionsResult{
+			Collections: res,
+		}, nil
```

Comment:
I think we could remove the duplication with the code below by re-setting the `list` variable and falling down.

Let's do the same in other similar places

Reply:
done. refactored all places.